### PR TITLE
Fix undetectable mock annotations with attributes above.

### PR DESCRIPTION
**Pull request by sidepelican**
mockolo cannot detect the mock annotation in code like the following:

```swift
@MainActor
/// @mockable
protocol P {
}
```
This PR fixes this issue.

### CODE REVIEW

File: Sources/MockoloFramework/Parsers/SwiftSyntaxExtensions.swift [hunk 1]
```diff
@@ -259,7 +259,7 @@ extension IfConfigDeclSyntax {
             in: subModels,
             exclude: [:],
             fullnames: []
-        ).map({ $0 })
+        ).sorted(path: \.value.offset, fallback: \.key)
```
Comment (sidepelican) [inline Sources/MockoloFramework/Parsers/SwiftSyntaxExtensions.swift:262]: Additionally, fix the fragile test caused by https://github.com/uber/mockolo/pull/279 .

Comment (sidepelican) [reply Sources/MockoloFramework/Parsers/SwiftSyntaxExtensions.swift:262]: This operation is same as  https://github.com/uber/mockolo/blob/1fb44ed25e962b52e18f293201a192375f9e018a/Sources/MockoloFramework/Operations/UniqueModelGenerator.swift#L69 .

File: Sources/MockoloFramework/Parsers/SwiftSyntaxExtensions.swift [hunk 2]
```diff
@@ -300,7 +300,7 @@ extension ProtocolDeclSyntax: EntityNode {
     }
 
     func annotationMetadata(with annotation: String) -> AnnotationMetadata? {
-        return leadingTrivia.annotationMetadata(with: annotation)
+        return leadingTrivia.annotationMetadata(with: annotation) ?? protocolKeyword.leadingTrivia.annotationMetadata(with: annotation)
```
Comment (fummicc1) [inline Sources/MockoloFramework/Parsers/SwiftSyntaxExtensions.swift:303]:
```suggestion
        let trivias = [
            leadingTrivia,
            protocolKeyword.leadingTrivia,
            modifiers.leadingTrivia
        ] + attributes.map(\.leadingTrivia)
        return trivias.lazy.compactMap { $0.annotationMetadata(with: annotation) }.first
```

I found this is not sufficient. Below is a failure pattern.

```swift
@MainActor
/// @mockable
public protocol P {
    func f() -> String
}
```

**Review by fummicc1 — CHANGES_REQUESTED** [review]
Thank you very much! I found one case that should be supported.

File: Sources/MockoloFramework/Parsers/SwiftSyntaxExtensions.swift (hunk 2 above)
Comment (sidepelican) [reply Sources/MockoloFramework/Parsers/SwiftSyntaxExtensions.swift]: Good catch! thank you.

**Review by fummicc1 — APPROVED** [review]
LGTM! Thank you for fixing fragile test as well. :)

**Pull request merged**